opensearchapi/opensearchapi.go

```diff
@@ -99,11 +101,34 @@ func NewDefaultClient() (*Client, error) {
 
 	return clientInit(rootClient), nil
 }
+func getOperationName() string {
+	pc, _, _, ok := runtime.Caller(1)
+	if ok {
+		fullName := runtime.FuncForPC(pc).Name()
+		return strings.ToLower(fullName)
+	}
+	return "unknown"
+}
+
 
 // do calls the opensearch.Client.Do() and checks the response for openseach api errors
-func (c *Client) do(ctx context.Context, req opensearch.Request, dataPointer any) (*opensearch.Response, error) {
+func (c *Client) do(providedCtx context.Context, req opensearch.Request, dataPointer any) (*opensearch.Response, error) {
+	
+	var ctx context.Context
+	if instrument := c.Client.InstrumentationEnabled(); instrument!=nil {
+		ctx = instrument.Start(providedCtx, getOperationName())
```

Comment:
This won't work for nested operations like `/_cluster/allocation/explain`, which should be `cluster.allocation_explain`  instead of `AllocationExplain`